### PR TITLE
Feature/relative time

### PR DESCRIPTION
**Pull request by MatthiasGu**

### CODE REVIEW

File: src/serina.ts [hunk 1]
```diff
@@ -15,7 +16,7 @@ const serina = (text: string): ParsedSchema => {
         matches: [],
     };
 
-    const filters = [ WeekDay, Day, Month, Year, Time, Dates, PartialDates, DateAndTime ];
+    const filters = [ WeekDay, Day, Month, Year, Time, Dates, PartialDates, DateAndTime, RelativeTime];
```
Comment (h-dong) [inline src/serina.ts]: You deleted my space

File: src/filters/time/relativeTime.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { ParsedMatchSchema } from 'serina.schema';
```
Comment (h-dong) [inline src/filters/time/relativeTime.ts]: Can we move this to a different folder, it is a different filter after all.

File: src/filters/time/time.constants.ts [hunk 3]
```diff
@@ -8,8 +8,46 @@ const TIME = {
...
+const SECONDS = '(seconds|second|secs|sec)';
+const TIME_UNITS = {
+    ANY: `(${HOURS}|${MINUTES}|${SECONDS})`,
+    HOURS: `${HOURS}`,
+    MINUTES: `${MINUTES}`,
+    SECONDS: `${SECONDS}`,
+};
+
+const HALF = '(half an|half a)';
+const QUARTER = '(a quarter of a|a quarter of an)';
```
Comment (h-dong) [inline src/filters/time/time.constants.ts]: what about "go to work 20 to 5" or "go to work 20 mins to 5" or "go to work 20 mins to 5pm"?

You are gonna tell me I'm scope creeping again lol

File: src/filters/time/time.constants.ts [hunk 4]
```diff
@@ -8,8 +8,46 @@ const TIME = {
...
+    HALF: `${HALF}`,
+    QUARTER: `${QUARTER}`,
+    ONE: `${ONE}`,
+};
+
+const ARGUMENT = `([0-9]+|${VERBAL_QUANTIFIERS.ANY}) ${TIME_UNITS.ANY}`;
+const ARGUMENT_AFTER = `${RELATIVE_PREPOSITIONS}${ARGUMENT}`;
+const ARGUMENT_FIRST = `${ARGUMENT}${RELATIVE_POSTPOSTIONS}`;
+
+const RELATIVE_TIME = {
```
Comment (h-dong) [inline src/filters/time/time.constants.ts]: Can we also move this to a different folder as it is relative time logic. And go back to return TIME as default export?

File: src/filters/time/time.constants.ts (hunk 3 above)
Comment (MatthiasGu) [reply src/filters/time/time.constants.ts]: That is not really relative time though?

Comment (h-dong) [reply src/filters/time/time.constants.ts]: I guess it was missed out of `Time`

Comment (h-dong) [reply src/filters/time/time.constants.ts]: So do you want to add it? If not I'll create an issue to track it

Comment (MatthiasGu) [reply src/filters/time/time.constants.ts]: I think this should be separate as it will require rewriting the time parser, so it is not a quick change.

Comment (MatthiasGu) [reply src/filters/time/time.constants.ts]: I can even take it myself but maybe not part of this PR

**Review by h-dong — APPROVED** [review]
